### PR TITLE
Fix #6853: schema for ImpactT

### LINKED ISSUE #6853: Create app schema for ImpactT
We should add initial support for the ImpactT code via the lume-impact python interface.

A sirepo schema should be create which includes lattice elements (drift, quadrupole, solenoid, etc) and simulation setup (integrator, mesh, distribution).

Refer to these repos:
https://github.com/impact-lbl/IMPACT-T/blob/master/examples/Sample1/ImpactT.in

https://github.com/ChristopherMayes/lume-impact/blob/master/docs/examples/input_parsing_example.ipynb

### PR DESCRIPTION
**Pull request by gurhar1133**
Used the manual to create the beam parameters and elements

### CODE REVIEW

File: sirepo/package_data/static/json/impactt-schema.json [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+            "n_z": ["Nz", "Integer", 32, "Number of mesh points in z"],
+            "flag_bc": ["Flagbc", "Integer", 1, "Field boundary condition flag: Currently must be set to 1 which corresponds to an open boundary condition."],
+            "x_rad": ["Xrad", "Float", 0.015, "Size of computational domain"],
+            "y_rad": ["Yrad", "Float", 0.015, "Size of computational domain"],
+            "perd_len": ["Perdlen", "Float", 45.0, "Size of computational domain"],
+            "flag_dist": ["FlagDist", "DistType", "Gaussian", "Type of the initial distribution."],
+            "r_start_flg": ["Rstartflag", "Integer", 0, "If restartflag = 1, restart the simulation from the previous check point. If restartflag = 0, start the simulation from the beginning."],
+            "n_emission": ["Nemission", "Integer", 400, "The number of numerical emission steps"],
+            "t_emission": ["Temission [sec]", "Float", 6.5e-12, "Temission Laser pulse emission time"],
+            "distribution_parameters": {
```
Comment (moellep) [inline sirepo/package_data/static/json/impactt-schema.json:63]: Move the "distribution_parameters" to the outer level, and you could call it "command_distribution". Nested model definitions aren't supported.

File: sirepo/package_data/static/json/impactt-schema.json [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+            "config": {
+                "controller": "impacttSourceController as source",
+                "templateUrl": "/static/html/ipmactt-source.html"
+            }
+        }
+    },
+    "model": {
+        "simulationStatus": {},
+        "impacttReport": {},
+        "beam": {
```
Comment (moellep) [inline sirepo/package_data/static/json/impactt-schema.json:39]: Rename "beam" to "command_beam" so we'll be able to use it on the standard lattice command editor.

File: sirepo/package_data/static/json/impactt-schema.json [hunk 3]
```diff
@@ -0,0 +1,206 @@
...
+            },
+            "b_curr": ["Bcurr [A]", "Float", 0.7140000000000003, "Beam current in Amps."],
+            "b_k_energy": ["Bkenergy [eV]", "Float", 1.0, "Initial beam pseudo-kinetic energy in eV."],
+            "b_mass": ["Bmas [eV]", "Float", 511005.0, "Mass of the particles in eV"],
+            "b_charge": ["Bcharge", "Float", -1.0,"Particle charge in units of proton charge."],
+            "b_freq": ["Bfreq [Hz]", "Float", 2856000000.0, "Reference frequency in Hz"],
+            "t_ini": ["Tini", "Float", -3.249141278122655e-12, "Initial reference time in seconds."]
+        },
+        "_ELEMENT": {
+            "zedge": ["Z Edge", "Float", 0, "Longitudinal position of the entrance face of the element"],
```
Comment (moellep) [inline sirepo/package_data/static/json/impactt-schema.json:94]: Add a "name" field to _ELEMENT, even though it isn't required in the generated output - that way items can be named within the standard lattice editor.

Also all elements have a length field "L" which is the first line in the input row.

File: sirepo/package_data/static/json/impactt-schema.json [hunk 4]
```diff
@@ -0,0 +1,206 @@
...
+            "b1_gradient": ["Gradient [T/m]", "Float", 0],
+            "L_effective": ["Effective Length [m]", "Float", 0]
+        },
+        "DIPOLE": {
+            "zedge": ["Z Edge", "Float", 0, "Longitudinal position of the entrance face of the element"],
+            "x_field_strength": ["X Field Strength [T]", "Float", 0],
+            "y_field_strength": ["Y Field Strength [T]", "Float", 0],
+            "half_gap_width": ["Half of Gap Width (m)", "Float", 0],
+            "fmapfn": ["FMAPFN", "InputFile", "", "file ID to contain the geometry information of bend."]
+        },
```
Comment (moellep) [inline sirepo/package_data/static/json/impactt-schema.json:144]: DIPOLE also has misalignments and rotations. 
We want to match the lume-impact naming conventions as much as possible, so the field names should be (b_field_x, b_field, filename, and half_gap)
Try the lume-impact parser to see, or refer to their parser implementation:
https://github.com/ChristopherMayes/lume-impact/blob/master/impact/parsers.py

**Review by moellep — CHANGES_REQUESTED** [review]

File: sirepo/package_data/static/json/impactt-schema.json (hunk 4 above)
Comment (gurhar1133) [reply sirepo/package_data/static/json/impactt-schema.json:146]: on page 13 of the [impact manual](https://portal.nersc.gov/project/m669/IMPACT-T/documents/ImpactTv2.pdf) it says misalignment and rotation is not used for dipole. 

It also looks like in lume-impact's `parse_dipole(line)` those values dont get parsed. 

I will work on the name consistency with lume-impact

**Comment by gurhar1133** [comment]
I went through the lume-impact parsers and the impact manual and made some changes to make names and fields more consistent with lume-impact. LMK what you think @moellep

**Review by moellep — APPROVED** [review]

**Pull request merged**